src/pages/IdeaDetail.jsx

```diff
@@ -2,31 +2,38 @@ import { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
 import useWorkshop from "../hooks/use-workshop";
 import Button from "../components/Button";
+import LoadingSpinner from "../components/LoadingSpinner";
 
 export default function IdeaDetail() {
   const { id } = useParams();
   const { workshop, isLoading, error } = useWorkshop(id);
   const [formattedCost, setFormattedCost] = useState("");
+  const [isOpenText, setIsOpenText] = useState("");
 
   useEffect(() => {
-    const formatCost = () => {
-      if (workshop && workshop.est_cost) {
-        const cost = workshop.est_cost;
-        const formattedCost = new Intl.NumberFormat("en-AU", {
-          style: "currency",
-          currency: "AUD",
-        }).format(cost);
-        setFormattedCost(formattedCost);
-      }
-    };
+    if (workshop) {
+      const formatCost = () => {
+        if (workshop.est_cost) {
+          const cost = workshop.est_cost;
+          const formattedCost = new Intl.NumberFormat("en-AU", {
+            style: "currency",
+            currency: "AUD",
+          }).format(cost);
+          setFormattedCost(formattedCost);
```

Comment:
formattedCost can be just a variable, instead of using useState.
can be 
```

const formattedCost = formatCost();
const formatConst = () => {
let formattedCost = "";
if (workshop && workshop.est_cost) {
        const cost = workshop.est_cost;
        formattedCost = new Intl.NumberFormat("en-AU", {
          style: "currency",
          currency: "AUD",
        }).format(cost);
 return formattedCost
}
```

And this way you can use the formatted cost variable where you need.